File: python/src/wdl_model/model/sections/wdl_command.py
"""Command section node for Python WDL tasks.

The WDL specification describes the command section as the template evaluated after inputs are
available and before outputs are evaluated. The stored string literal preserves placeholders used
for interpolation.
"""

from __future__ import annotations

from dataclasses import dataclass

from wdl_model.model.definitions import WdlTaskElement
from wdl_model.model.expressions import WdlStringLiteral


@dataclass
class WdlCommand(WdlTaskElement):
    """Task command template and its multiline/single-line representation."""

    commandText: WdlStringLiteral | None = None
    multiline: bool = False

    def getCommandText(self) -> WdlStringLiteral | None:
        """Return the command template literal."""
        return self.commandText

    def setCommandText(self, commandText: WdlStringLiteral) -> None:
        """Set the command template literal."""
        self.commandText = commandText

    def isMultiline(self) -> bool:
        """Return whether the source used the multiline command form."""
        return self.multiline

    def setMultiline(self, multiline: bool) -> None:
        """Set whether the source used the multiline command form."""
        self.multiline = multiline

File: typescript/src/model/sections/wdl-command.ts
/** Command section nodes. */
import type { WdlTaskElement } from '../definitions/wdl-task.js';
import type { WdlStringLiteral } from '../expressions/wdl-string-literal.js';

/** Models a task `command` section and its multiline/single-line form. */
export class WdlCommand implements WdlTaskElement {
  /** Creates a command section from its template literal and multiline flag. */
  public constructor(
    private commandTextValue?: WdlStringLiteral,
    private multilineValue = false,
  ) {}

  /** Returns the command template literal. */
  public getCommandText(): WdlStringLiteral | undefined {
    return this.commandTextValue;
  }
  /** Sets the command template literal. */
  public setCommandText(commandText: WdlStringLiteral | undefined): void {
    this.commandTextValue = commandText;
  }
  /** Returns whether the command used the multiline `<<< >>>` form. */
  public isMultiline(): boolean {
    return this.multilineValue;
  }
  /** Sets whether the command used the multiline `<<< >>>` form. */
  public setMultiline(multiline: boolean): void {
    this.multilineValue = multiline;
  }
}
